pkg/test-infra/cdc/operation.go

```diff
@@ -16,193 +16,227 @@ package cdc
-		cc.Sts.Spec.Replicas = desiredSts.Spec.Replicas
-		cc.Sts.Spec.PodManagementPolicy = desiredSts.Spec.PodManagementPolicy
-		return nil
+
+	if err := c.waitCDCReady(cc.StatefulSet, 5*time.Minute); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Ops) waitCDCReady(st *appsv1.StatefulSet, timeout time.Duration) error {
+	local := st.DeepCopy()
+	log.Infof("Waiting up to %v for StatefulSet %s to have all replicas ready",
+		timeout, st.Name)
+	return wait.PollImmediate(5*time.Second, timeout, func() (bool, error) {
+		key, err := client.ObjectKeyFromObject(local)
+		if err != nil {
+			return false, err
+		}
+
+		err = c.cli.Get(context.TODO(), key, local)
+		if err != nil && errors.IsNotFound(err) {
+			return false, err
+		}
+		if err != nil {
+			log.Errorf("error getting cdc statefulset: %v", err)
+			return false, err
+		}
+		if local.Status.ReadyReplicas != *local.Spec.Replicas {
+			log.Infof("StatefulSet %s found but there are %d ready replicas and %d total replicas.",
+				local.Name, local.Status.ReadyReplicas, *local.Spec.Replicas)
+			return false, nil
+		}
+		log.Infof("All %d replicas of StatefulSet %s are ready.", local.Status.ReadyReplicas, local.Name)
+		return true, nil
 	})
-	return cc, err
 }
 
-func (c *CdcOps) DeleteCDC(cc *CDC) error {
-	err := c.cli.Delete(context.TODO(), cc.Sts)
-	if err != nil && !errors.IsNotFound(err) {
+func (c *Ops) waitJobCompleted(job *batchv1.Job, timeout time.Duration) error {
+	local := job.DeepCopy()
+	log.Infof("Waiting up to %s for job completed", local.Name)
+
+	return wait.PollImmediate(5*time.Second, timeout, func() (bool, error) {
+		key, err := client.ObjectKeyFromObject(local)
+		if err != nil {
+			return false, err
+		}
+
+		err = c.cli.Get(context.TODO(), key, local)
+		if err != nil && errors.IsNotFound(err) {
+			return false, err
+		}
+		if err != nil {
+			log.Errorf("error getting cdc job: %v", err)
+			return false, err
+		}
+
+		if local.Status.Succeeded != 1 {
+			log.Infof("Job %s was not completed.", local.Name)
+			return false, nil
+		}
+
+		log.Infof("Job %s has been completed.", local.Name)
+		return true, nil
+	})
+}
+
+// ApplyObject applies object
+func (c *Ops) ApplyObject(object runtime.Object) error {
+	err := c.cli.Create(context.TODO(), object)
+	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
 	return nil
 }
 
-func (c *CdcOps) StartJob(job *CDCJob, spec *CDCSpec) error {
-	kjob, err := c.renderSyncJob(job, spec)
-	if err != nil {
+// Delete cdc component
+func (c *Ops) DeleteCDC(cc *CDC) error {
+	err := c.cli.Delete(context.TODO(), cc.StatefulSet)
+	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
+	return nil
+}
 
-	if err := c.cli.Create(context.TODO(), kjob); err != nil {
+//  ApplyJob applies cdc job
+func (c *Ops) ApplyJob(job *batchv1.Job) error {
+	if err := c.ApplyObject(job); err != nil {
 		return err
 	}
-
-	return nil
+	return c.waitJobCompleted(job, 5*time.Second)
```

Comment:
5s maybe too short here?

Reply:
And I think it' unnecessary to parameterize it, maybe you can use `fixture.Context.WaitClusterReadyDuration` in waitJobCompleted directly.